### ISSUE TITLE
Issue with optimizes probability threshold Function 

### ISSUE THREAD

**Issue by gm039** [pycaret/pycaret#2128]
**Describe the bug**

optimize_threshold is not working as expected. I am using PyCaret '2.3.6'

**To Reproduce**

## PyCaret Version
import pycaret
pycaret.__version__

'2.3.6'
## Predict using a trained model
predict_model(Model, raw_score=True)
print(Model)

![image](https://user-images.githubusercontent.com/50217085/152604304-f25c76fc-0058-47df-b4de-49a82b44defe.png)

## Optimize threshold and predict using the threshold optimized model
Model_Thres_Opt = optimize_threshold(Model, optimize='f1')
print(Model_Thres_Opt)
predict_model(Model_Thres_Opt, raw_score=True)

![image](https://user-images.githubusercontent.com/50217085/152605211-e41d1661-c712-4462-8231-9fa7d7abc25f.png)

![image](https://user-images.githubusercontent.com/50217085/152604424-652bddd9-580d-4849-9d71-b11a37229e3d.png)

**Expected behavior**

I expect the F1 value of optimized model should be 0.56 at probability threshold=0.1, as can be seen from the figure.
Also, the value of AUC should be same regardless of probability threshold value. Can you please help in resolving it?

</details>

**Comment by moezali1** [comment]
@gm039 The difference between cross-validation and hold-out metric could be due to overfitting. 

I checked if `predict_model` has any bug and it is ignoring the threshold for scoring. I couldn't find any bug.

```
from pycaret.datasets import get_data
data = get_data('juice')

from pycaret.classification import *
s = setup(data, target = 'Purchase', session_id = 123)

dt = create_model('dt')

optimized_dt = optimize_threshold(dt)

predict_model(optimized_dt);

predict_model(dt);
```
![image](https://user-images.githubusercontent.com/54699234/153734008-05c3ebe1-b4f6-4101-a526-2c25f717ac33.png)

**Comment by gm039** [comment]
Thanks @moezali1 

I see the difference in probability scores of model after using optimize_threshold. See attached snap.
I believe that the first 3 columns should have same values. Any comments?

# import dataset and setup
from pycaret.datasets import get_data
data = get_data('juice')

from pycaret.classification import *
s = setup(data, target = 'Purchase', session_id = 1234,silent=True)

# train a model
Model = create_model('rf')
print(Model)
Pred=predict_model(Model);


# optimize threshold
optimized_Model = optimize_threshold(Model)
Pred_Opt=predict_model(optimized_Model);
print(optimized_Model)

# optimize threshold for F1
optimized_Model_f1 = optimize_threshold(Model,optimize='F1')
Pred_Opt_F1=predict_model(optimized_Model_f1)
print(optimized_Model_f1)

# use raw score to compare
Pred_Opt_F1_Raw=predict_model(optimized_Model_f1, raw_score=True)


# print prob. scores
pd.concat([Pred['Score'], Pred_Opt['Score'],Pred_Opt_F1['Score'],Pred_Opt_F1_Raw['Score_CH'],Pred_Opt_F1_Raw['Score_MM']], axis=1)
![juice](https://user-images.githubusercontent.com/50217085/155256276-0f3aaff7-e621-435f-a347-c9cc159f0676.PNG)

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
